cpp/ql/src/JPL_C/LOC-3/Rule 17/BasicIntTypes.ql

```diff
@@ -38,8 +39,8 @@ Type getAUsedType(Type t) {
 }
 
 predicate problematic(IntegralType t) {
-  // List any exceptions that should be allowed.
-  any()
+  // 'bool' is allowed as it represents a 'true' or 'false' value
+  t.getName() != ["bool"]
```

Comment:
We had some internal discussion as to whether we should allow `bool`. We converged allowing it, but the story is a bit muddled here, as JPL C and Misra C 2004 both assume you're using some pre-C11 version of C, which does not have either a `bool` or `_Bool` type.

I do think this needs to be solved a bit differently, and should not depend on `getName`, because if your code was targeting an older version of C, then `bool` might potentially represent something totally differently. My suggestion is to leave this predicate as it was, and changed the `where` clause below to include `not usedType instanceof BoolType`.

Reply:
This makes sense to me.  Thanks for the advice!